lib/pages/guide.js

```diff
@@ -79,13 +70,47 @@ function reindexFootnoteListing(footnotesDiv, footnoteMap) {
 }
 
 function reindexFootnotes(bodyHast) {
-  const footnotesDiv = bodyHast.children.filter(child => child.tagName === 'div' && child.properties.className.includes('footnotes'))[0];
-  if(!footnotesDiv) { throw new Error(`No footnotes div found. Did you call unified.use(footnotes)?`)};
+  const footnotesDiv = bodyHast.children.filter(child => child.tagName === 'div' && child.properties.className && child.properties.className.includes('footnotes'))[0];
+  if(!footnotesDiv) {
+    throw new Error(`No footnotes div found. Did you call unified.use(footnotes)?`);
+  };
 
   const footnoteMap = reindexFootnoteMarkersUnderNode(bodyHast);
   reindexFootnoteListing(footnotesDiv, footnoteMap);
 }
 
+function htmlDirectives() {
+  return (tree) => {
+    visit(tree, 'image', configureImage);
+    visit(tree, 'blockquote', configureBlockquotes);
+  };
+
+  function configureImage(node) {
+    const hast = h('image', {
+      ...node,
+      className: "inline",
+      align: 'right',
+    });
+    
+    node.data = {
+      hName: hast.tagName,
+      hProperties: hast.properties,
+    };
```

Comment:
I don't think I understand what you're getting out of doing it this way. This results in HTML that looks like:
```html
<img
  src="images/3dr-diagram.png"
  alt="Three Dumb Routers layout diagram"
  title="Image showing layout of the three dumb routers system."
  type="image"
  url="images/3dr-diagram.png"
  position="[object Object]"
  class="inline"
  align="right"
>
```

- The `align="right"` is pretty well deprecated at this point; we should use `float: right` in the CSS instead.
- The rest of the valuable properties (alt, title) come through automatically.
- I can't see what you're trying to do with the type, url, or position properties.

If you change this so it just says:
```js
function configureImage(node) {
  node.data = { hProperties: { className: 'inline' } }; // or whatever className you want
}
```

Then you get a much nicer:
```html
<img
  src="images/3dr-diagram.png"
  alt="Three Dumb Routers layout diagram"
  title="Image showing layout of the three dumb routers system."
  class="inline"
>
```

No cruft!

Reply:
That said, I do think that this would be easier with a generic class-adding directive (`:attrs{.diagram}` maybe?), though, so that we don't hit every image in the page (assuming we might add more).

Reply:
Or you could use unist-util-visit-parents and only apply this to images that are inside of a blockquote.

Reply:
I didn't understand how the visit-parents was supposed to work